rskj-core/src/test/java/co/rsk/peg/BridgeTest.java

```diff
@@ -773,37 +731,35 @@ public void activeAndRetiringFederationOnly_activeFederationIsNotFromFederateMem
                 null
         );
 
-        ActivationConfig activations = spy(ActivationConfigsForTest.genesis());
-        doReturn(true).when(activations).isActive(eq(RSKIP143), anyLong());
+        int senderPK = 405; // Sender PK belongs to retiring federation member PKs
+        Integer[] activeMemberPKs = new Integer[]{100, 200, 300, 400, 500, 600};
+        Integer[] retiringMemberPKs = new Integer[]{101, 202, 303, 404, 505, 606};
 
-        Federation activeFederation = new Federation(
-                FederationTestUtils.getFederationMembersFromPks(100, 200, 300, 400, 500, 600),
-                ZonedDateTime.parse("2017-06-10T02:30:01Z").toInstant(),
-                0L,
-                NetworkParameters.fromID(NetworkParameters.ID_REGTEST)
-        );
-        Federation retiringFederation = new Federation(
-                FederationTestUtils.getFederationMembersFromPks(101, 202, 303, 404, 505, 606),
-                ZonedDateTime.parse("2017-06-10T02:30:01Z").toInstant(),
-                0L,
-                NetworkParameters.fromID(NetworkParameters.ID_REGTEST)
-        );
+        Federation activeFederation = FederationTestUtils.getFederation(activeMemberPKs);
+        Federation retiringFederation = FederationTestUtils.getFederation(retiringMemberPKs);
+
+        Bridge bridge = getBridgeInstance(activeFederation, retiringFederation, senderPK);
+
+        // When
+        executor.execute(bridge, null);
+
+        // Then
+        verify(decorate, times(1)).execute(any(), any());
+    }
 
+    private Bridge getBridgeInstance(Federation activeFederation, Federation retiringFederation, int senderPK) {
         BridgeSupport bridgeSupportMock = mock(BridgeSupport.class);
         doReturn(activeFederation).when(bridgeSupportMock).getActiveFederation();
         doReturn(retiringFederation).when(bridgeSupportMock).getRetiringFederation();
 
-        // RSK Address: FederationTestUtils -> ECKey.fromPrivate(BigInteger.valueOf(PK + 1))
-        ECKey key = ECKey.fromPrivate(BigInteger.valueOf(405));
+        ECKey key = ECKey.fromPrivate(BigInteger.valueOf(senderPK));
         Transaction rskTxMock = mock(Transaction.class);
         doReturn(new RskAddress(key.getAddress())).when(rskTxMock).getSender();
 
-        Bridge bridge = getBridgeInstance(rskTxMock, bridgeSupportMock, activations);
-
-        executor.execute(bridge, null);
+        ActivationConfig activations = spy(ActivationConfigsForTest.genesis());
+        doReturn(true).when(activations).isActive(eq(RSKIP143), anyLong());
 
-        verify(bridgeSupportMock, times(1)).getActiveFederation();
```

Comment:
where is this verification?

Reply:
This was removed since the mocks are now inside the build method. We can leave it outside and add the verification again.

Reply:
it'd be nice

Reply:
Done!